### PR TITLE
Improve shortcut management

### PR DESCRIPTION
**Pull request by sclassen**

### CODE REVIEW

File: core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java [hunk 1]
```diff
@@ -145,51 +143,44 @@ public void createWindowsMenu() throws IOException {
...
+            LOG.debug("Fallback to reading Shortcuts with default encoding {}", Charset.defaultCharset().name());
+            return Files.readAllLines(shortcutFile.toPath(), Charset.defaultCharset());
+        }
+    }
+
+    private boolean needToAddNewShortcutEntry(String path, List<String> lines) {
+        return lines.stream()
+                .map(line -> line.split(","))
+                .map(array -> array[1])
+                .anyMatch(sPath -> sPath.equalsIgnoreCase(path));
```
Comment (janakmulani) [inline core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java]: noneMatch - if we don't find a match then we need to add

Comment (sclassen) [reply core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java]: thanks, fixed

File: core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java [hunk 2]
```diff
@@ -145,51 +143,44 @@ public void createWindowsMenu() throws IOException {
...
-            }
-            if (fAdd) {
-                LOG.debug("Default encoding is {}", Charset.defaultCharset().name());
-                LOG.debug("Adding Shortcut to list = {} with UTF-8 encoding", sItem);
-                String scInfo = file.getFileLocation().toString() + ",";
-                scInfo += path + "\r\n";
-                Files.write(getWindowsShortcutsFile().toPath(), scInfo.getBytes(UTF_8), StandardOpenOption.APPEND);
-            }
+        LOG.debug("Using WindowsShortCutManager {}", shortcutFile);
+        final List<String> lines = readAllLine(shortcutFile);
```
Comment (AndreasEhret) [inline core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java]: wouldn't be "readAllLines" a better method name here?

Comment (sclassen) [reply core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java]: full ack

**Pull request merged**